module/zfs/zfs_log.c

```diff
@@ -626,6 +629,10 @@ zfs_log_write(zilog_t *zilog, dmu_tx_t *tx, int txtype,
 		off += len;
 		resid -= len;
 	}
+
+	if (write_state == WR_COPIED || write_state == WR_NEED_COPY) {
+		dsl_pool_wrlog_delay(size, zilog->zl_dmu_pool, tx->tx_txg);
+	}
```

Comment:
Unconditionally blocking here on the txg sync may introduce some subtle problems since it will result in `zfs_write()` holding open its transaction handle until the txg sync completes.  That can be problematic since if the pool gets suspended and the txg can't be written there's no way to unwind the in progress write.

A better place to wait (and kick) would be in `dmu_tx_assign()` which is already set up to delay when we're over the dirty limit and on error all callers are setup to unwind any open transactions.  However, updating the wrlog accounting there would be tricky so it probably makes good sense to leave that bit here.